### ISSUE TITLE
Add more context

### ISSUE THREAD

**Issue by adamshakhabov** [alefragnani/vscode-bookmarks#311]
_I have not much experience using this extension and maybe it already exists. If not, then what about:_

Add the opportunity to catch more context for a bookmark. For example, as a developer, I
1) select function;
2) make a bookmark;
3) as today, function name will be bookmark's subject;
4) function body (with some limit) will be displayed when hovering a cursor over the link bookmark.

**Comment by adamshakhabov** [comment]
@alefragnani, I could not clone `vscode-bookmarks-core` submodule. Is it possible to contribute this project?

**Comment by adamshakhabov** [comment]
@alefragnani, I could not clone `vscode-bookmarks-core` submodule. Is it possible to contribute this project?

**Comment by adamshakhabov** [comment]
@alefragnani, I could not clone `vscode-bookmarks-core` submodule. Is it possible to contribute this project?

**Comment by alefragnani** [comment]
Hi @adamshakhabov ,

> when hovering a cursor over the link bookmark

What do you mean about that? Are you talking about the Tooltip displayed in the Side Bar?

If that’t the case, I think is doable to create a `context` config which you decide the _number of lines_ (up / down) to be displayed in the Tooltip. 

On the other hand, using _function bodies_ or anything like that wouldn’t be available, basically because it would be tied to languages (to correctly recognize language-related structures).

Hope this helps

**Comment by alefragnani** [comment]
About cloning/contributions to the repo, thank you for your interest, but it’s not possible right now. 

I’m refactoring a few pieces of the extension and until this work is done, it will be closed.

Thank you

**Comment by adamshakhabov** [comment]
> Are you talking about the Tooltip displayed in the Side Bar?

Yes, I mean Tooltip which can be configured as you suggested above by some new `context` variable.

**Comment by dseynhae** [comment]
I'm not sure what the original intention for this issue was, but I was about to request this feature:

When hovering over a bookmark in the code margin, the label content should appear (like a tooltip). If no label was used, then simply display `Bookmark`.

This is very similar to breakpoints (which are in the same margin): if you hover over those, you will see the expression, or  `Breakpoint`/`Disabled Breakpoint`.

Since this is probably what this issue is about, I just wanted to be a bit more explicit about the request.

Note: there currently is a "tooltip" in the Side Bar. That doesn't seem to make much sense to me, as it just displays the label that we already see? There is a different request to extend bookmarks into sticky notes... It would make a lot more sense if the tooltips in the Side Bar would display the content of the sticky notes...

**Comment by alefragnani** [comment]
Hi @dseynhae ,

The idea is to provide more than the _current line_ as tooltip on each bookmark, displaying about 3 lines above and 3 below. Doing so, the user has an idea of a _preview/magnifying glass_.

Your request, on the other hand, unfortunately is not possible yet, because VS Code does not provide an API for extensions to display anything related to its icon added to the gutter. There is already an open issue on their repo, almost 5 years old (https://github.com/microsoft/vscode/issues/28080), and not ETA.

Hope this helps

**Comment by dseynhae** [comment]
That's really a pity (no API to get information from objects in the gutter).

Since breakpoints already do this, it must be easy for Microsoft to create an API for it... but then, if it really was easy, we probably would already have it.

**Comment by alefragnani** [comment]
I'm pretty sure it is doable, because the feature already exists (as you said, breakpoint has). It is just a matter of _exposing an API_. To sum up, they recently added Context Menu and drag & drop support on gutter, but only for their own features. New APIs on the _same location/component_.

BTW, that issue has only 14 upvotes, which means that not so many extension developers needs that, unfortunately. VS Code issue tracker today closes issues with less than 20 upvotes after 60 days, if I remember correctly. So, it's a matter of time (or their anual clean up milestone) to that issue to be closed.

I think it is just a matter of _priority_, and just 14 upvotes means a lot in this case. And even if such process blocks me from adding features to my extensions, I totally understand them. I do the same for my extensions.
